rskj-core/src/main/java/co/rsk/peg/federation/ErpFederation.java

```diff
@@ -74,13 +75,26 @@ public Script getRedeemScript() {
                     getNumberOfEmergencySignaturesRequired(),
                     activationDelay
                 );
+
+                validateScriptSize();
             } catch (RedeemScriptCreationException e) {
                 throw new ErpFederationCreationException(e.getMessage(), e, REDEEM_SCRIPT_CREATION_FAILED);
             }
         }
         return redeemScript;
     }
 
+    private void validateScriptSize() {
+        if (getFormatVersion() != P2SH_P2WSH_ERP_FEDERATION.getFormatVersion()) {
+            // since the redeem script is located in the script sig,
+            // we need to check it does not surpass the maximum allowed size
+            ScriptValidations.validateScriptSigElementSize(redeemScript);
+            return;
+        }
+
+        ScriptValidations.validateWitnessScriptSize(redeemScript, getNumberOfSignaturesRequired());
```

Comment:
Are we sure the limit is impossed to the witness script or is it to the redeem script?

Reply:
Updated.